### PR TITLE
AESH Upgrade and CLI refactoring

### PR DESCRIPTION
**Pull request by Gonthim**

### CODE REVIEW

**Comment by bstansberry** [comment]
Good to see. I'm not going to comment much on this as it's at a level where only @aloubyansky 's expertise will do. :)

File: pom.xml [hunk 1]
```diff
@@ -95,7 +95,7 @@
         <version.org.codehaus.woodstox.stax2-api>3.1.4</version.org.codehaus.woodstox.stax2-api>
         <version.org.codehaus.woodstox.woodstox-core-asl>4.4.1</version.org.codehaus.woodstox.woodstox-core-asl>
         <version.org.fusesource.jansi>1.9</version.org.fusesource.jansi>
-        <version.org.jboss.aesh>0.33.16</version.org.jboss.aesh>
+        <version.org.jboss.aesh>0.62</version.org.jboss.aesh>
```
Comment (bstansberry) [inline pom.xml]: This doesn't matter for this PR and my comment is off topic, but this is a weird versioning scheme. This project is already being used in high value products with long, long support cycles, so what's the point of aesh still being on major version 0? Given the changes required to use this new version, it's clear 0.6x is not really part of the same major series as 0.3x.

**Comment by Gonthim** [comment]
retest this please

Removed the test commit. I'll PR it later since a multi-commit PR didn't work like I expected.

**Comment by Gonthim** [comment]
These errors don't make any sense. The missing method that's causing most of these, CLITestUtil line 70, shows as being removed in the commit. Going to close this and create a new branch to pull it again.

**Comment by Gonthim** [comment]
These errors don't make any sense. The missing method that's causing most of these, CLITestUtil line 70, shows as being removed in the commit. Going to close this and create a new branch to pull it again.

**Pull request closed without merging**